### PR TITLE
fix(table): pagination.current display error when result of filter is empty

### PR DESCRIPTION
**Pull request by jinchaofs**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/29364

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Table `pagination.current` display error when result of filter is empty.        |
| 🇨🇳 Chinese |  修复表格当过滤结果为空时，`pagination.current` 展示错误。  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by jinchaofs** [comment]
忘记link issue了：https://github.com/ant-design/ant-design/issues/29364

**Comment by afc163** [comment]
来个用例

**Comment by jinchaofs** [comment]
通过Ant Design 官网的Demo可以复现：
1.打开链接：https://ant.design/components/table-cn/#components-table-demo-head
2.对Name列进行筛选，筛选结果为空，如下图所示：

3.点击Name列排序，查看调试工具，打印结果 current 为 0，如下图所示：


总结下来，问题原因是，无论是筛选还是过滤，分页会被重置，但是在判断maxPage的时候忽略了为0的情况，导致current会出现为0的情况

**Comment by afc163** [comment]
补一下测试用例

**Comment by jinchaofs** [comment]
@afc163 测试用例已添加：https://github.com/ant-design/ant-design/pull/29481

**Comment by jinchaofs** [comment]
> 补一下测试用例

已补充

**Comment by afc163** [comment]
这个用例貌似没改动 components/table/hooks/usePagination.ts 也是过的？

https://github.com/ant-design/ant-design/runs/1976719329

**Comment by jinchaofs** [comment]
> 这个用例貌似没改动 components/table/hooks/usePagination.ts 也是过的？
> 
> https://github.com/ant-design/ant-design/runs/1976719329

😌应该是测试用例的问题，我再改下

**Comment by jinchaofs** [comment]
> > 这个用例貌似没改动 components/table/hooks/usePagination.ts 也是过的？
> > https://github.com/ant-design/ant-design/runs/1976719329
> 
> 😌应该是测试用例的问题，我再改下

@afc163 确实是测试用例有问题，导致重现不了Bug，已修复

**Comment by jinchaofs** [comment]
@afc163 有人帮忙review吗😄

**Comment by yoyo837** [comment]
应该都下班回家了，莫慌，`antd`的PR在github里算处理得比较快的了。

File: components/table/__tests__/Table.filter.test.js [hunk 1]
```diff
@@ -1508,4 +1508,53 @@ describe('Table.filter', () => {
...
+
+    const wrapper = mount(
+      <Table onChange={onChange} rowKey="name" columns={columns} dataSource={dataSource} />,
+    );
+    wrapper.find('.ant-dropdown-trigger').first().simulate('click');
+    wrapper.find('FilterDropdown').find('MenuItem').at(0).simulate('click');
+    wrapper.find('.ant-btn-primary').first().simulate('click');
+    expect(onChange.mock.calls[0][0].current).toBe(1);
+
+    wrapper.find('.ant-dropdown-trigger').first().simulate('click');
```
Comment (kerm1it) [inline components/table/__tests__/Table.filter.test.js:1555]: 这里是不是只需要测试一次就行了？第一次结果就为空了吧？

File: components/table/hooks/usePagination.ts [hunk 2]
```diff
@@ -71,7 +71,8 @@ export default function usePagination(
   // Reset `current` if data length or pageSize changed
   const maxPage = Math.ceil((paginationTotal || total) / mergedPagination.pageSize!);
```
Comment (kerm1it) [inline components/table/hooks/usePagination.ts:72]:
```suggestion
  const maxPage = Math.max(Math.ceil((paginationTotal || total) / mergedPagination.pageSize!), 1);
```

这样是不是更严谨一点？

**Review by kerm1it — APPROVED** [review]

**Pull request merged**